### ISSUE TITLE
Request: Fold based on explicit markers

### ISSUE THREAD

**Issue by dfh** [gregsexton/origami.el#13]
I would find it very useful to be able to fold on explicit fold markers, à la folding.el! For example in long configuration files with no indentation. I think explicit markers are the only viable solution then? Anyway, my elisp is not quite sharp enough to add it myself without making a big mess, I'm afraid. But I'm hoping that maybe it's quite an easy fix?

Thanks!

**Comment by gregsexton** [comment]
Thanks for adding this issue. I'll try to whip something up soon.

**Comment by Klinkenstecker** [comment]
A simple solution, that uses hardcoded markers (in this case `//{` for the start of a fold and `//}` for the end of a fold) into the parser looks like this:
```elisp
(defun origami-marker-parser (create)
  (lambda (content)
    (let ((positions (origami-get-positions content "//{\\|//}")))
      (origami-build-pair-tree create "//{" "//}" positions))))
```

That works already, but fold also the text in the line behind `//{`, which is a different to folding.el, and therefore not so great for my code base. My elisp is also really bad, but i hacked a solution that you can find here: 
https://gist.github.com/Klinkenstecker/8e21c8b36b2ddd91a073

**Comment by gregsexton** [comment]
Hi. Sorry for the delay, I've really let github issues and PRs pile up on me. Taking a look at this and associated PR.

**Comment by gregsexton** [comment]
Closed #17
